src/compas_timber/elements/fastener.py

```diff
@@ -19,65 +19,88 @@ class Fastener(TimberElement):
+    shape : :class:`~compas.geometry.Geometry`, optional
         The geometry of the fastener.
-    frame : :class:`~compas.geometry.Frame`
-        The frame of the fastener.
+    transformation : :class:`~compas.geometry.Transformation`, optional
+        The transformation of the fastener.
+    **kwargs : dict, optional
+        Additional keyword arguments.
 
     Attributes
     ----------
-    geometry : :class:`~compas.geometry.Geometry`
+    shape : :class:`~compas.geometry.Geometry`
         The geometry of the fastener.
-    frame : :class:`~compas.geometry.Frame`
-        The frame of the fastener.
+    transformation : :class:`~compas.geometry.Transformation`
+        The transformation of the fastener.
+    interfaces : list
+        A list of interfaces associated with this fastener.
+    attributes : dict
+        Dictionary of attributes for this fastener.
+    debug_info : list
+        A list of debug information.
+    is_fastener : bool
+        Always True for fasteners.
+    key : int or None
+        The graph node key of this fastener.
 
     """
 
-    def __init__(self, shape=None, frame=None, **kwargs):
-        super(Fastener, self).__init__(**kwargs)
+    def __init__(self, shape=None, transformation=None, **kwargs):
+        super(Fastener, self).__init__(transformation=transformation, **kwargs)
         self._shape = shape
         self.interfaces = []
-        self.frame = frame
         self.attributes = {}
         self.attributes.update(kwargs)
         self.debug_info = []
 
     def __repr__(self):
         # type: () -> str
-        return "Fastener(frame={!r}, name={})".format(self.frame, self.name)
+        return "Fastener(transformation={!r}, name={})".format(self.transformation, self.name)
 
     def __str__(self):
         # type: () -> str
         return "<Fastener {}>".format(self.name)
 
-    @property
-    def frame(self):
-        return self._frame
-
-    @frame.setter
-    def frame(self, frame):
-        self._frame = frame
-
     @property
     def is_fastener(self):
+        """Check if this element is a fastener.
+
+        Returns
+        -------
+        bool
+            Always True for fasteners.
+        """
         return True
 
     @property
     def key(self):
+        """The graph node key of this fastener.
+
+        Returns
+        -------
+        int or None
+            The graph node key, or None if not set.
+        """
         # type: () -> int | None
         return self.graphnode
 
     @property
     def __data__(self):
         return {
             "shape": self._shape,
-            "frame": self.frame,
+            "transformation": self.transformation,
             "interfaces": self.interfaces,
         }
 
     def compute_geometry(self):
-        """returns the geometry of the fastener in the model"""
-        return self.shape.transformed(Transformation.from_frame(self.frame))
+        """Returns the geometry of the fastener in the model.
```

Comment:
should this be maybe `compute_elementgeometry`? meaning this should implement `Element`s interface